dashboard/state_aggregator.py

```diff
@@ -25,6 +26,41 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass(init=True)
+class StateApiResult:
+    # Returned data.
+    data: Union[dict, List[dict], Dict[str, JobInfo]] = None
+    # A list of warnings generated from the API that should be delivered to users.
+    warnings: List[str] = None
+
+
+GCS_QUERY_FAILURE_WARNING = (
+    "Failed to query data from GCS. It is due to "
+    "(1) GCS is unexpectedly failed. "
+    "(2) GCS is overloaded by lots of work. "
+    "(3) There's an uexpected network issues. Please check the GCS logs to "
+    "find the root cause."
+)
+RAYLET_QUERY_FAILURE_WARNING = (
+    "Failed to query data from some raylets. You might have data loss. "
```

Comment:
"You might have data loss.". Does it mean we lost data or incomplete results?

Reply:
It is incomplete result. But I feel like it is also data loss because within this snapshot, we lost the state data? What are the differences between incomplete result vs data loss?

Reply:
data loss usually means permanent lost of data due to unrecoverable failures and an overkill. I think incomplete result might be more accurate.